src/program/services/downloaders/alldebrid.py

```
@@ -116,60 +116,29 @@ def _validate_premium(self) -> bool:
-        result = {}
-        for i, (name, size) in enumerate(self._walk_files(files)):
-            if (
-                any(name.lower().endswith(ext) for ext in VIDEO_EXTENSIONS)
-                and "sample" not in name.lower()
-            ):
-                result[str(i)] = {"filename": name, "filesize": size}
-        return result
+        finally:
+            if torrent_id:
+                self.delete_torrent(torrent_id)
+            return return_value
```

Ruff (0.8.0)

`return` inside `finally` blocks cause exceptions to be silenced

(B012)